### PR TITLE
test(eval,tool,skill,prompt): make weak assertions bite

### PR DESCRIPTION
**Pull request by kojiwakayama**
## Summary

Audit of all 44 test files under `src/eval`, `src/tool`, `src/prompt` and `src/skill` for tests that stay green when the behavior they claim to cover breaks.

The bar for every change: name a concrete source edit that breaks documented behavior yet leaves the test passing. Every change was **mutation checked** — the named edit was applied, the test confirmed to fail, then the edit reverted.

65 candidate findings, 56 confirmed after adversarial verification, 55 applied, 9 refuted. **No product defects found. Test files only.**

## Recurring weaknesses fixed

**Compound conditions tested from one side only.** A judge whose pass is `modelPass && score >= threshold` was only ever exercised with *both* true, so `pass: modelPass` alone passed every test. Both halves are now pinned, plus the exact boundary case, so `>=` cannot silently become `>`.

**Clamping tested at one end.** Score clamping asserted only the upper bound, so `return Math.min(1, score)` passed. The lower bound is now pinned too.

**Generated artifacts asserted by substring.** The JUnit report is now asserted **by value** including both `<failure>` children — one carrying an explanation, one carrying stringified evidence — so deleting the failure loop fails. The markdown report now asserts the example status row and the regression status line rather than merely that a file was written.

**Blocking severity was inferred rather than observed.** A budget metric and a soft metric now run together in one case with the record outcomes asserted separately, so dropping `|| result.severity === "budget"` from the blocking check fails.

**Resource bounds were undefended.** Skill frontmatter parsing now asserts the depth, per-container and total-node caps each reject — with a **positive control** so the limits cannot be lowered to zero to make the tests pass. That positive control is the part that matters; without it, gutting the limits is a valid way to make bound tests green.

**Input validation was not proven to run before dispatch.** Bounded document parsing now asserts the provider call counter stays at **zero** when content is rejected, so the guard cannot be moved after the call.

**Constraint arms that no test reached:** cost above max, groundedness below min, and groundedness not measured at all.

## One suggested case replaced

The audit proposed a `passRate { min: 0.9 }` constraint case. That arm is unreachable through this fixture — any `passRate` below 1 sets `summary.failed > 0`, which short-circuits before constraints are evaluated. Substituted `groundednessScore` for the min arm instead of writing a test that could never exercise the branch.

## Verification

- Semantic unit-boundary audit: ok
- Sanitizer ratchet test: ok, `test:layout`: ok, `docs:api-reference:check`: current
- `deno fmt`, `deno lint`: pass
- **25/25** changed test files pass
- `anti-slop`, `cwd-relative-test-reads`, `test-typecheck`, `check-awaits`, `cross-runtime-jsr`, `skipped-tests`, `ban-test-only`: all pass

No test was deleted, skipped, or weakened.

### CODE REVIEW

**Review by kwakayama — CHANGES_REQUESTED** [review]
> *This was generated by AI during triage.*\n\n## Review findings\n\n[MEDIUM]  and : unbounded promises can hang the suite if cancellation or the injected wait regresses. Add bounded failure paths. [MEDIUM] : replace the tight wall-clock upper bound with deterministic clock/timer or behavior assertions.

**Review by kwakayama — CHANGES_REQUESTED** [review]
> *This was generated by AI during triage.*

## Review findings

[MEDIUM] `src/prompt/factory.test.ts:239` and `src/tool/sleep.test.ts:50`: unbounded promises can hang the suite if cancellation or the injected wait regresses. Add bounded failure paths. [MEDIUM] `src/eval/agent-service-hardening.test.ts:1112`: replace the tight wall-clock upper bound with deterministic clock/timer or behavior assertions.

**Review by kwakayama — CHANGES_REQUESTED** [review]
> *This was generated by AI during triage.*\n\n[P2] Three new tests can hang or flake: cancellation awaits before releasing the generator gate, waitCalled is unbounded, and polling asserts a short wall-clock upper bound. Release gates before awaiting and use bounded/deterministic synchronization.

**Review by kwakayama — DISMISSED** [review]
> *This was generated by AI during triage.*\n\nRe-reviewed the pushed head and addressed review threads. The verified fixes are present and the targeted regression checks reported by the implementation pass. Approved pending required repository checks.

**Review by kwakayama — DISMISSED** [review]
> *This was generated by AI during triage.*\n\nRe-reviewed the new pushed head. The follow-up finding is addressed, the relevant review threads are resolved, and focused tests passed. Approved pending required checks.

File: src/tool/project-scoped-remote-tools.test.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import "#veryfront/schemas/_test-setup.ts";
-import { assertEquals, assertStrictEquals, assertThrows } from "@std/assert";
+import { it } from "#veryfront/testing/bdd.ts";
+import { assertEquals, assertRejects, assertStrictEquals, assertThrows } from "@std/assert";
```
[the remark this answers is not in the document]
Comment (kojiwakayama) [reply src/tool/project-scoped-remote-tools.test.ts]: Fixed. Line 3 was `import { assertEquals, assertRejects, assertStrictEquals, assertThrows } from "@std/assert";` — this PR had added `assertRejects` to that pre-existing direct import while switching the file to #veryfront/testing/bdd.ts. All four symbols now come from "#veryfront/testing/assert.ts" (src/testing/assert.ts exports all four; assertRejects at line 306). No assertion was weakened or removed; all 24 cases in the file still pass.
